specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Patches.json

```diff
@@ -0,0 +1,357 @@
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/patches": {
+      "get": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_ListByBuilderResource",
+        "description": "List PatchResource resources by BuilderResource.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "builderName",
+            "in": "path",
+            "required": true,
+            "description": "The name of the builder",
+            "minLength": 2,
+            "maxLength": 32,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/PatchCollection"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/patches/{patchName}": {
+      "get": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_Get",
+        "description": "Get a PatchResource.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./Builders.json#/parameters/BuilderNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PatchNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patches_Get_0": {
+            "$ref": "./examples/Patches_Get.json"
+          }
+        }
+      },
+      "put": {
```

Comment:
Do we even expect customers to create a patch resource themselves? Since it is a proxy resource, we do not need to have a PUT request, as the RP can create the resources without going through ARM. Take [container app revisions](https://github.com/Azure/azure-rest-api-specs/blob/release-Microsoft.App-2023-05-02-preview/specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/ContainerAppsRevisions.json) for example. They do not need a PUT (create or update) request since they are created by default when the container app is, or when customer make [a revision scope change](https://learn.microsoft.com/en-us/azure/container-apps/revisions#revision-scope-changes) in their container app.

So for patching, I imagine that we would create potential patches behind the scenes with the timed function, and then customers would be able to `apply` the patch, similar to how customers can [`activate` a revision](https://github.com/Azure/azure-rest-api-specs/blob/release-Microsoft.App-2023-05-02-preview/specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/ContainerAppsRevisions.json#L128).

Reply:
Makes sense! This one is now updated.

Reply:
I also noticed that you deleted the `delete` command for patches. Do we want to keep customers from deleting patches that they know they will never apply?